### PR TITLE
update version support tables

### PR DESCRIPTION
**Pull request by tuminoid**
Update version support tables with information about ironic-image tags and releases.

### CODE REVIEW

**Comment by tuminoid** [comment]
> /hold
> 
> Ironic-image 25.0 just came out, I'll need to edit that in tomorrow.

/unhold

Done

File: docs/user-guide/src/version_support.md [hunk 1]
```diff
@@ -86,16 +101,16 @@ The table describes which branches/image-tags are tested in each periodic CI tes
...
-| metal3-periodic-ubuntu/centos-e2e-feature-test-main            | main         | main        | main           | latest         | latest      | latest     |
-| metal3-periodic-ubuntu/centos-e2e-feature-test-release-1-7     | release-1.7  | release-1.7 | release-0.6    | v0.6.0         | latest      | latest     |
-| metal3-periodic-ubuntu/centos-e2e-feature-test-release-1-6     | release-1.6  | release-1.6 | release-0.5    | v0.5.0         | latest      | latest     |
-| metal3-periodic-ubuntu/centos-e2e-feature-test-release-1-5     | release-1.5  | release-1.5 | release-0.4    | v0.4.0         | latest      | latest     |
+| -------------------------------------------------------------- | ------------ | ----------- | -------------- | -------------- | ----------- | ------------ |
+| metal3-periodic-ubuntu/centos-e2e-feature-test-main            | main         | main        | main           | latest         | latest      | latest       |
+| metal3-periodic-ubuntu/centos-e2e-feature-test-release-1-7     | release-1.7  | release-1.7 | release-0.6    | v0.6.1         | latest      | release-24.1 |
```
Comment (Rozzii) [inline docs/user-guide/src/version_support.md:111]: I would rather but the "v" tags for Ironic image as a deliverable that we advertise, release-x image tags are moving tags.

Comment (tuminoid) [reply docs/user-guide/src/version_support.md]: done

File: docs/user-guide/src/version_support.md [hunk 2]
```diff
@@ -86,16 +101,16 @@ The table describes which branches/image-tags are tested in each periodic CI tes
...
 | metal3_periodic_main_integration_test_ubuntu/centos            | main         | main        | main           | latest         | latest      | latest     |
-| metal3-periodic-ubuntu/centos-e2e-integration-test-release-1-7 | release-1.7  | release-1.7 | release-0.6    | v0.6.0         | latest      | latest     |
-| metal3-periodic-ubuntu/centos-e2e-integration-test-release-1-6 | release-1.6  | release-1.6 | release-0.5    | v0.5.0         | latest      | latest     |
-| metal3-periodic-ubuntu/centos-e2e-integration-test-release-1-5 | release-1.5  | release-1.5 | release-0.5    | v0.5.0         | latest      | latest     |
+| metal3-periodic-ubuntu/centos-e2e-integration-test-release-1-7 | release-1.7  | release-1.7 | release-0.6    | v0.6.1         | latest      | v24.1.1          |
+| metal3-periodic-ubuntu/centos-e2e-integration-test-release-1-6 | release-1.6  | release-1.6 | release-0.5    | v0.5.0         | latest      | v24.0.0          |
+| metal3-periodic-ubuntu/centos-e2e-integration-test-release-1-5 | release-1.5  | release-1.5 | release-0.5    | v0.5.0         | latest      | v23.1.0          |
```
Comment (dtantsur) [inline docs/user-guide/src/version_support.md:106]: The table renders correctly but the last column is now misaligned in the source. Do we care?

Comment (tuminoid) [reply docs/user-guide/src/version_support.md:106]: Whoops, I'll fix it.

Comment (tuminoid) [reply docs/user-guide/src/version_support.md]: Done. It was linewrapping in the editor, and looked aligned. Now it should actually be aligned.

**Comment by tuminoid** [comment]
OK, cleaned up the tables after switch from branches to tag, so they could be made narrower + aligned. PTAL.

File: docs/user-guide/src/version_support.md [hunk 3]
```diff
@@ -57,6 +57,21 @@ Following table summarizes BMO release/test process:
 | v0.2          | EOL       |
 | v0.1          | EOL       |
 
+## Ironic-image
+
+Since `v23.1.0`, Ironic follows the semantic versioning scheme for its own
+release cycle, the same way as CAPM3 and IPAM. Currently, we have release-24.1, release-24.0
+and release-23.1 release branches for `v24.1.x` `v24.0.x` and `v23.1.x` release cycle
```
Comment (dtantsur) [inline docs/user-guide/src/version_support.md:64]: Do you want to mention 25.0 and its release cycle as well?

Comment (tuminoid) [reply docs/user-guide/src/version_support.md]: I put it in the table, but forgot the text part, as 25.0 came out just after initial version of the PR. Good catch, and fixed.
The text is now saying `two or three`  as we need to maintain any number needed for CAPM3 releases on Metal3, even though Ironic bugfix branch itself would not be maintained.

**Review by Rozzii — COMMENTED** [review]
/lgtm

**Pull request merged**